test/e2e/devices/devicegettest/device_get_test.go

```diff
@@ -0,0 +1,91 @@
+package devicestest
+
+import (
+	"io"
+	"os"
+	"strings"
+	"testing"
+
+	root "github.com/equinix/metal-cli/internal/cli"
+	"github.com/equinix/metal-cli/internal/devices"
+	outputPkg "github.com/equinix/metal-cli/internal/outputs"
+	"github.com/equinix/metal-cli/test/helper"
+	"github.com/spf13/cobra"
+)
+
+func TestCli_Devices_Get(t *testing.T) {
+	var projectId, deviceId string
+	var err error
+	var resp bool
+	subCommand := "device"
+	consumerToken := ""
+	apiURL := ""
+	Version := "metal"
+	rootClient := root.NewClient(consumerToken, apiURL, Version)
+	type fields struct {
+		MainCmd  *cobra.Command
+		Outputer outputPkg.Outputer
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		want    *cobra.Command
+		cmdFunc func(*testing.T, *cobra.Command)
+	}{
+		{
+			name: "get_by_device_id",
+			fields: fields{
+				MainCmd:  devices.NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				root := c.Root()
+				projectId, err = helper.CreateTestProject("metal-cli-get-pro")
+				if err != nil {
+					t.Error(err)
+				}
+				deviceId, err = helper.CreateTestDevice(projectId, "metal-cli-get-dev")
+				if err != nil {
+					t.Error(err)
+				}
+				root.SetArgs([]string{subCommand, "get", "-i", deviceId})
+				rescueStdout := os.Stdout
+				r, w, _ := os.Pipe()
+				os.Stdout = w
+				if err := root.Execute(); err != nil {
+					t.Error(err)
+				}
+				w.Close()
+				out, _ := io.ReadAll(r)
+				os.Stdout = rescueStdout
+				if !strings.Contains(string(out[:]), deviceId) {
+					t.Error("expected output should include " + deviceId)
+				}
+				if len(projectId) != 0 && len(deviceId) != 0 {
+					resp, err = helper.IsDeviceStateActive(deviceId)
+					if err == nil && resp == true {
+						err = helper.CleanTestDevice(deviceId)
+						if err != nil {
+							t.Error(err)
+						}
+						err = helper.CleanTestProject(projectId)
+						if err != nil {
+							t.Error(err)
+						}
+					} else {
+						t.Error(err)
```

Comment:
Latest test run failed here with:
    `device_get_test.go:77: <nil>`

It looks like we need a couple things:
- `CreateTestDevice` should return a non-nil error if the device is still not active after the retry loop is done
- We probably need a longer wait time for the device to become active.